### PR TITLE
Update sklearn-onnx to work with onnx 1.7

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

File: .azure-pipelines/linux-conda-CI.yml [hunk 1]
```diff
@@ -113,7 +129,13 @@ jobs:
           python setup.py install
           cd ..
         else
-          pip install onnx$(onnx.version)
+          if [ '$(onnx.version)' == 'test' ]
```
Comment (wenbingl) [inline .azure-pipelines/linux-conda-CI.yml:132]: A trick of the test pypi issue is to install a official release, then upgrade to the test package.

File: skl2onnx/common/_topology.py [hunk 2]
```diff
@@ -1140,7 +1153,12 @@ def _get_main_opset_version(model):
     """
     Returns the main opset version.
     """
+    mld = None
     for op in model.opset_import:
         if op.domain == '':
```
Comment (wenbingl) [inline skl2onnx/common/_topology.py:1158]: isn't ai.onnx the official name, is it?

File: skl2onnx/proto/__init__.py [hunk 3]
```diff
@@ -61,4 +61,19 @@ def make_tensor_fixed(name, data_type, dims, vals, raw=False):
...
+
+def get_latest_tested_opset_version():
+    """
+    This module relies on *onnxruntime* to test every
+    converter. The function returns the most recent
+    target opset tested with *onnxruntime* or the opset
+    version specified by *onnx* package if this one is lower
+    (return by `onnx.defs.onnx_opset_version()`).
+    """
+    return min(11, get_opset_number_from_onnx())
```
Comment (wenbingl) [inline skl2onnx/proto/__init__.py:79]: can we use the number from onnxconvert-common as well, so we can minimize the change on the opset upgrading?

File: .azure-pipelines/linux-conda-CI.yml (hunk 1 above)
Comment (xadupre) [reply .azure-pipelines/linux-conda-CI.yml:132]: I won't probably be able to cherry pick. I'll modify the CI file when onnx 1.7 is released.

File: skl2onnx/common/_topology.py (hunk 2 above)
Comment (xadupre) [reply skl2onnx/common/_topology.py:1158]: It is left blank in the onnx graph. With older version of onnx, it was failing with ai.onnx.

File: skl2onnx/proto/__init__.py (hunk 3 above)
Comment (xadupre) [reply skl2onnx/proto/__init__.py:79]: upgrading this number is onnxconverter-common implies every converter library was able to switch to the next opset. I prefer that every conveter defines the latest opset it is compatible with.

File: skl2onnx/algebra/graph_state.py [hunk 4]
```diff
@@ -143,25 +143,25 @@ def _ty_astype(cst):
         elif isinstance(cst, np.int64):
             name = self.scope.get_unique_variable_name(
                 self.onnx_prefix + 'cst')
-            ty = AttributeProto.INT
+            ty = TensorProto.INT64  # AttributeProto.INT
```
[the remark this answers is not in the document]
Comment (xadupre) [reply skl2onnx/algebra/graph_state.py:146]: Sorry, I forgot.

File: skl2onnx/algebra/graph_state.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
 # --------------------------------------------------------------------------
 import numpy as np
 from scipy.sparse import coo_matrix
-from onnx import AttributeProto
+# from onnx import AttributeProto
```
[the remark this answers is not in the document]
Comment (xadupre) [reply skl2onnx/algebra/graph_state.py]: Fixed

File: skl2onnx/algebra/graph_state.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (xadupre) [reply skl2onnx/algebra/graph_state.py]: Fixed.

File: skl2onnx/operator_converters/gaussian_process.py [hunk 6]
```diff
@@ -90,9 +90,18 @@ def convert_gaussian_process_regressor(scope, operator, container):
         mean_y = op._y_train_mean.astype(dtype)
         if len(mean_y.shape) == 1:
             mean_y = mean_y.reshape(mean_y.shape + (1,))
-        y_mean = OnnxAdd(y_mean_b, mean_y,
-                         output_names=out[:1],
+        y_mean = OnnxAdd(y_mean_b, mean_y, output_names=out[:1],
                          op_version=opv)
+        # if not hasattr(op, '_y_train_std') or op._y_train_std == 1:
```
Comment (prabhat00155) [inline skl2onnx/operator_converters/gaussian_process.py:95]: Commented out code here, can you remove it?

File: tests/test_algebra_cascade.py [hunk 7]
```diff
@@ -43,6 +43,11 @@ def generate_onnx_graph(dim, nbnode, input_name='X1', opv=None):
                np.array([[93., 93., 93., 93., 93.]]),
                np.array([[100100., 100100., 100100., 100100., 100100.]])]
         for opv in ({'': 10}, None, 9, 10, 11, onnx_opset_version()):
+            if isinstance(opv, dict):
+                if opv[''] > get_latest_tested_opset_version():
+                    continue
+            elif opv is not None and opv > get_latest_tested_opset_version():
```
Comment (prabhat00155) [inline tests/test_algebra_cascade.py:49]: You may just write `opv` instead of `opv is not None`.

Comment (xadupre) [reply tests/test_algebra_cascade.py:49]: It is not exactly the same. bool([] is not None) is True and bool([]) is False.

File: skl2onnx/operator_converters/gaussian_process.py (hunk 6 above)
Comment (xadupre) [reply skl2onnx/operator_converters/gaussian_process.py]: Done.

**Comment by xadupre** [comment]
I reply to an earlier comment about the number of changed files, the change I had to consider is every unit test was previously using onnx opset by default but opset 12 is not supported by onnxruntime 1.2. I had to change the default opset version used for testing.

File: tests/test_algebra_cascade.py (hunk 7 above)
Comment (prabhat00155) [reply tests/test_algebra_cascade.py:49]: Yes, but opv isn't a list here. It's either None or int value. Also, even if we had list here, wouldn't you want a bool([]) to be considered False?

Comment (xadupre) [reply tests/test_algebra_cascade.py:49]: True, it is an integer but the comment is the same.

File: .azure-pipelines/win32-conda-CI.yml [hunk 8]
```diff
@@ -13,47 +13,69 @@ jobs:
     vmImage: 'vs2017-win2016'
   strategy:
     matrix:
+      Py37-Onnx170-Rt120-0230:
+        python.version: '3.7'
+        # onnx.version: '-i https://test.pypi.org/simple/ onnx'
```
Comment (prabhat00155) [inline .azure-pipelines/win32-conda-CI.yml:18]: Remove the commented line please.

File: tests/test_sklearn_gaussian_process.py [hunk 9]
```diff
@@ -453,14 +453,34 @@ def test_gpr_rbf_unfitted(self):
...
         reason="onnxruntime %s" % THRESHOLD)
-    def test_gpr_rbf_fitted(self):
+    def test_gpr_rbf_fitted_true(self):
 
         gp = GaussianProcessRegressor(alpha=1e-7,
                                       n_restarts_optimizer=15,
                                       normalize_y=True)
+        gp, X = fit_regression_model(gp)
+
+        # return_cov=False, return_std=False
```
Comment (prabhat00155) [inline tests/test_sklearn_gaussian_process.py:467]: Why is this comment here?

**Pull request merged**